src/command_modules/azure-cli-network/azure/cli/command_modules/network/commands.py

```diff
@@ -331,16 +332,32 @@ def _make_singular(value):
     with self.command_group('network dns record-set') as g:
         g.custom_command('list', 'list_dns_record_set', client_factory=cf_dns_mgmt_record_sets, transform=transform_dns_record_set_output)
 
-    for record in ['a', 'aaaa', 'mx', 'ns', 'ptr', 'srv', 'txt', 'caa']:
+    api_version = str(get_api_version(self.cli_ctx, ResourceType.MGMT_NETWORK_DNS))
+    api_version = api_version.replace('-', '_')
```

Comment:
@tjprescott does CLI core have any built function to accomplish this? The above two lines code seem generic to me.

Reply:
I don't believe so.